examples/benchmark/benchmark_mnist_shared_hyper.yaml

```diff
@@ -0,0 +1,82 @@
+# test benchmark config.
+mode: grid
+test_domains:
+  - 3
+  - 0
+
+output_dir: zoutput/benchmarks/mnist_benchmark
+# number of hyperparameter samples per task.
+# Thus, the total runs of each task are given
+# by len(domains) * num_param_samples * num_seeds (see below)
+num_param_samples: 8
+sampling_seed: 0
+
+startseed: 0
+endseed: 2  # currently included
+
+
+Shared params:
+    gamma_y:
+      min: 1e4
+      max: 1e5
+      step: 100
+      distribution: loguniform
+      num: 2
+
+    zy_dim:
+      min: 32
+      max: 96
+      step: 32
+      distribution: uniform
+      num: 2
+
+
+domainlab_args:
+  # Domainlab arguments passed to each task.
+  # task specific arguments take precedence.
+  ## dataset
+  task: mnistcolor10
+  tr_d: [1, 2, 4]
+  dmem: False
+  lr: 0.001
+  epos: 50
+  es: 5
+  bs: 64
+  nname: conv_bn_pool_2
+  san_check: True
+
+
+method_diva:  # name
+  aname: diva
+  zx_dim: 0
+  zd_dim: 32
+  shared:
+    - gamma_y
+    - zy_dim
+  nname_dom: conv_bn_pool_2
+
+  hyperparameters:
+    gamma_d:
+      min: 1e4
+      max: 1e6
+      step: 10_000
+      distribution: loguniform
+      num: 2
+    zx_dim:
+      min: 0
+      max: 96
+      step: 32
+      distribution: uniform
+      num: 2
+
+
+method_hduva:  # name
+  aname: hduva
+
+  nname_topic_distrib_img2topic: conv_bn_pool_2
+  nname_encoder_sandwich_layer_img2h4zd: conv_bn_pool_2
+  shared:
+    - gamma_y
+    - zy_dim
+  zx_dim: 0
+  zd_dim: 32
```

Comment:
the current yaml file which does not specify any hyper-parameters for hduva, will not generate shared parameters for hduva. 

7,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 64}"
58,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 32}"
59,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 64}"
60,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 32}"
61,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 64}"
62,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 32}"
63,method_diva,diva,"{'gamma_d': 1000000.0, 'zx_dim': 64, 'gamma_y': 100000.0, 'zy_dim': 64}"
64,method_hduva,hduva,{}